orchestrator/crates/orchestrator/src/worker/event_handler/triggers/batching.rs

```
@@ -0,0 +1,278 @@
+                        storage
+                            .put_data(Bytes::from(serde_json::to_string(&state_update)?), &squashed_state_updates_path)
+                            .await?;
+                        // Add the new batch info in the database
+                        database
+                            .create_batch(Batch::create(batch.index + 1, block_number, squashed_state_updates_path))
+                            .await?;
+                        batch_index = batch.index + 1;
+                    }
+                } else {
+                    // No batch exists in the DB yet
+                    // Create a fresh batch
+
+                    let squashed_state_updates_path = self.get_state_update_file_name(batch_index + 1);
+                    // Put the state update in storage
+                    storage
+                        .put_data(Bytes::from(serde_json::to_string(&state_update)?), &squashed_state_updates_path)
+                        .await?;
+                    // Add the new batch info in the database
+                    database.create_batch(Batch::create(1, block_number, squashed_state_updates_path)).await?;
+                    batch_index = 1;
+                }
+                tracing::info!("Completed batching for block {}. Assigned batch {}", block_number, batch_index);
+            }
+            PendingUpdate(_) => {
+                tracing::info!("Skipping batching for block {} as it is still pending", block_number);
+            }
+        }
+
+        Ok(())
+    }
+
+    fn get_state_update_file_name(&self, batch_index: u64) -> String {
+        format!("{}/batch/{}.json", STATE_UPDATE_DIR, batch_index)
+    }
+
+    /// squash_state_updates merge all the StateUpdate into a single StateUpdate
+    fn squash_state_updates(&self, state_updates: Vec<StateUpdate>) -> Result<StateUpdate, JobError> {
+        if state_updates.is_empty() {
+            panic!("Cannot merge empty state updates");
```

let's throw an error in case of this

---

+1

---

done